core/src/services/config.js

```diff
@@ -1,85 +1,94 @@
-import { isFunction, isPromise } from '../utilities/helpers.js';
+import { getConfigValueFromObjectAsync } from '../utilities/async-helpers';
+import { getConfigValueFromObject } from '../utilities/helpers';
 
-(function() {
-  window.Luigi = {
-    setConfig: configInput => {
-      window.Luigi.config = configInput;
-    },
-    getConfig: () => {
-      return window.Luigi.config;
-    }
-  };
-})();
-/*
- * Gets value of the given property on the given object.
- */
-export const getConfigValueFromObject = (object, property) => {
-  let propIndex = 0;
-  let nextValue = object;
-  const propertyPath = property.split('.');
-  while (nextValue && propIndex < propertyPath.length) {
-    nextValue = nextValue[propertyPath[propIndex++]];
+class LuigiConfigManager {
+  constructor() {
+    this.configReadyTimeout = {
```

Comment:
Do we need this object?
The `valueMs` is not used anywhere. My suggestion would be to change this `configReadyTimeout` from an object to a string with just the `id` value.
The other option would be to use this `valueMs` in the timeout below, but then I don't like the name of the object, because what is an id of a timeout then?

Reply:
Id of a timeout makes sense to me, it is the way you identify a timeout you set in case you want to do something with it, like removing it. I will use valueMs instead the magic number '2000' when setting the timeout and keep the object.